src/kv/kv.h

```diff
@@ -1694,6 +1694,22 @@ namespace kv
           h->append(rep.p, rep.n, data.data(), data.size());
         }
       }
+      else
+      {
+        // Transactions containing a pre prepare or a pbft request should not
+        // contain anything else
+        if (views.size() > 1)
+        {
+          LOG_FAIL_FMT("Unexpected contents in pbft transaction {}", v);
+          return DeserialiseSuccess::FAILED;
+        }
+
+        auto search = views.find("ccf.pbft.preprepares");
+        if (search != views.end())
+        {
+          success = DeserialiseSuccess::PASS_PRE_PREPARE;
+        }
```

Comment:
It feels like we should check that the one entry we have either a write set on request or on prepares and fail otherwise. Here, a pbft request could consist of a write set to a single table that's not either, and we'd silently take it.

Reply:
We will most likely add a view change table soon so we will have to add another check for it too. That being said I think it is a fair point however and it can't hurt to check. If it we silently took it here pbft would (or at least should) fail later on on a merkle tree root check.

Reply:
Failing early will be easier to debug, if it ever happens :)